sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -358,3 +358,101 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_missing_indicator_dense():
+    # one feature with all missng values
+    X = np.array([
+       [-1,  -1,   2,   3],
+       [4,  -1,   6,  -1],
+       [8,  -1,  10,  11],
+       [12,  -1,  -1,  15],
+       [16,  -1,  18,  19]
+    ])
+    n_samples, n_features = X.shape
+    imputer = Imputer(missing_values=-1, strategy='mean', axis=0)
+    imputer_with_in = Imputer(missing_values=-1, strategy='mean', axis=0,
+                              add_missing_indicator=True)
+    Xt = imputer.fit_transform(X)
+    Xt_with_in = imputer_with_in.fit_transform(X)
+    imputed_features_mask = X[:, imputer_with_in.imputed_features_] == -1
+    assert_array_equal(Xt_with_in, np.hstack((Xt, imputed_features_mask)))
```

Comment:
You should check both of the above two lines, it can be possible that both the attribute `imputed_features_` and `Xt_with_in` are wrong but both give the same output. More precisely

    
    assert_array_equal(imputer_with_in.imputed_features_, [0, 2, 3])
    f_mask = X[:, [0, 2, 3]] = -1
    assert_array_equal(Xt_with_in, np.hstack((Xt, f_mask))